### PR TITLE
add PreserveTimes option

### PR DESCRIPTION
**Pull request by siscia**
The `PreserveTimes` option allows the copy to maintains the mtime and ctime of files and directory.

### CODE REVIEW

File: copy.go [hunk 1]
```diff
@@ -47,7 +49,17 @@ func copy(src, dest string, info os.FileInfo, opt Options) error {
...
 	}
-	return switchboard(src, dest, info, opt)
+
+	err = switchboard(src, dest, info, opt)
+
+	if opt.PreserveTimes {
+		mtime := info.ModTime()
+		stat := info.Sys().(*syscall.Stat_t)
+		atime := time.Unix(int64(stat.Atim.Sec), int64(stat.Atim.Nsec))
+		os.Chtimes(dest, atime, mtime)
```
Comment (otiai10) [inline copy.go:59]: Should we handle error from `os.Chtimes`, even if it's less likely to happen?

**Review by otiai10 — CHANGES_REQUESTED** [review]
Thank you! It makes sense to me.
Can you add test cases for this feature?

File: copy.go (hunk 1 above)
Comment (siscia) [reply copy.go:59]: I am not sure, how would you handle it?

**Comment by siscia** [comment]
So, this was quite more difficult than anticipate.

Go does not have a way to set the time for symlink. I ended up with two different implementation for `_linux` and `_windows`, could not test the one for windows.

**Comment by otiai10** [comment]
Can I take your PR over and add some commits?

**Comment by siscia** [comment]
Definitely!

I don't mind at all!

I was trying to keep things as simple as possible, but I would leave to you the direction of the PR!

No problem at all!

**Comment by siscia** [comment]
Hello, do we got any updates?

**Comment by otiai10** [comment]
sorry I couldn't take time for this last week. I'll work on it hopefully this week 😖

**Comment by siscia** [comment]
Definitely not a problem! I was asking just for my internal scheduling :)

**Pull request merged**